Lombok.NET/DecoratorGenerator.cs

```diff
@@ -196,5 +154,44 @@ private static bool IsAccessModifier(SyntaxToken token)
 		       || token.IsKind(SyntaxKind.ProtectedKeyword)
 		       || token.IsKind(SyntaxKind.PrivateKeyword)
 		       || token.IsKind(SyntaxKind.InternalKeyword);
+
+	/// <summary>
+	/// This is mostly reimplementing SyntaxGenerator.MethodDeclaration (which I don't believe can be used in a source generator).
+	/// On top of that, it tweaks the modifiers to be an implementation delegates the call to the <paramref name="delegationTargetIdentifier"/>.
+	/// </summary>
+	public static MethodDeclarationSyntax GenerateMethodDeclaration(IMethodSymbol symbol, string delegationTargetIdentifier)
+	{
+		return MethodDeclaration(symbol.ReturnType.ToTypeSyntax(), symbol.Name)
+			.WithModifiers(GenerateImplementationModifiers(symbol))
+			.WithParameterList(symbol.Parameters.GenerateParameterList())
+			.WithBody(Block(GenerateDelegatingCall(symbol, delegationTargetIdentifier)));
 	}
+
+	private static StatementSyntax GenerateDelegatingCall(IMethodSymbol symbol, string delegationTargetIdentifier)
+	{
+		var invocation = InvocationExpression(
+			MemberAccessExpression(
+				SyntaxKind.SimpleMemberAccessExpression,
+				IdentifierName(delegationTargetIdentifier),
+				IdentifierName(symbol.Name)
+			)
+		).WithArgumentList(ArgumentList(SeparatedList(
+			symbol.Parameters.Select(p => Argument(IdentifierName(p.Name))
+				.WithRefKindKeyword(p.RefKind.GenerateRefKindToken()))
+		)));
+
+		return symbol.ReturnsVoid ?
+			ExpressionStatement(invocation) :
+			ReturnStatement(invocation);
+	}
+
+	private static SyntaxTokenList GenerateImplementationModifiers(IMethodSymbol symbol) => TokenList(
+		new[]
+		{
+			symbol.GenerateAccessibilityToken(),
+			symbol.IsStatic ? Token(SyntaxKind.StaticKeyword) : Token(SyntaxKind.None),
+			symbol.IsOverride || symbol.IsVirtual || symbol.IsAbstract ?
+				Token(SyntaxKind.OverrideKeyword) : Token(SyntaxKind.None)
+		}.Where(t => !t.IsKind(SyntaxKind.None)
+		));
```

Comment:
I kept these methods here rather than in SymbolExtensions since they felt slightly less general. Though it's a bit borderline, and they may be able to be rearranged to make more of the code generalised rather than delegation specific.